packages/components/src/FormField/FormFieldTypes.ts

```diff
@@ -29,6 +29,272 @@ export type AutocompleteTypes =
+   * @see {@link https://www.w3.org/TR/wai-aria-1.2/#aria-activedescendant}
+   */
+  readonly ariaActiveDescendant?: AriaAttributes["aria-activedescendant"];
+
+  /**
+   * Indicates the entered value does not conform to the format expected.
+   * Supports boolean or specific error types: "grammar" | "spelling".
+   * @see {@link https://www.w3.org/TR/wai-aria-1.2/#aria-invalid}
+   */
+  readonly ariaInvalid?: AriaAttributes["aria-invalid"];
+
+  /**
+   * Indicates the element that controls the current element.
+   * @see {@link https://www.w3.org/TR/wai-aria-1.2/#aria-controls}
+   */
+  readonly ariaControls?: AriaAttributes["aria-controls"];
+
+  /**
+   * Indicates whether the element is expanded or collapsed.
+   * @see {@link https://www.w3.org/TR/wai-aria-1.2/#aria-expanded}
+   */
+  readonly ariaExpanded?: AriaAttributes["aria-expanded"];
+
+  /**
+   * Indicates the type of autocomplete interaction.
+   * @see {@link https://www.w3.org/TR/wai-aria-1.2/#aria-autocomplete}
+   */
+  readonly ariaAutocomplete?: AriaAttributes["aria-autocomplete"];
+
+  /**
+   * Indicates that user input is required before form submission.
+   * @see {@link https://www.w3.org/TR/wai-aria-1.2/#aria-required}
+   */
+  readonly ariaRequired?: AriaAttributes["aria-required"];
+}
+
+/**
+ * Focus event handlers for input elements.
+ * Generic interface that can be specialized for different element types.
+ */
+export interface FocusEvents<Target = HTMLElement> {
+  /**
+   * Focus event handler.
+   */
+  readonly onFocus?: (event: React.FocusEvent<Target>) => void;
+
+  /**
+   * Blur event handler.
+   */
+  readonly onBlur?: (event: React.FocusEvent<Target>) => void;
+}
+
+/**
+ * Keyboard event handlers for input elements.
+ * Generic interface that can be specialized for different element types.
+ */
+export interface KeyboardEvents<Target = HTMLElement> {
+  /**
+   * Key down event handler.
+   */
+  readonly onKeyDown?: (event: React.KeyboardEvent<Target>) => void;
+
+  /**
+   * Key up event handler.
+   */
+  readonly onKeyUp?: (event: React.KeyboardEvent<Target>) => void;
+}
+
+/**
+ * Curated set of HTML input attributes for rebuilt input components.
+ * This provides a whitelist of standard HTML/React props we want to support,
+ * avoiding the issues of extending React.InputHTMLAttributes directly.
+ * Note: Event handlers and ARIA attributes are separate - use FocusEvents, KeyboardEvents, and AriaInputProps.
+ */
+export interface HTMLInputBaseProps extends AriaInputProps {
+  /**
+   * The unique identifier for the input element.
+   */
+  readonly id?: string;
+
+  /**
+   * The name attribute for the input element.
+   */
+  readonly name?: string;
+
+  /**
+   * Placeholder text that appears when the input is empty.
+   */
+  readonly placeholder?: string;
+
+  /**
+   * Whether the input is disabled.
+   */
+  readonly disabled?: boolean;
+
+  /**
+   * Whether the input is required.
+   */
+  readonly required?: boolean;
```

Comment:
debating adding a note here, that we often implement this in other ways in practice eg. RHF/Zod